src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -71,6 +82,11 @@
   private final AppContext appContext;
   private final ConfigOverridesApplier configOverridesApplier;
 
+  public static final String BATCH_METRICS_ENABLED_CONF_FILE = "batch_metrics_enabled.conf";
+  private boolean batchMetricsEnabled;
+  private boolean defaultBatchMetricsEnabled = false;
+  private ConcurrentSkipListSet<Long> batchMetricsDBSet;
+
```

Comment:
`batchMetricsDBSet` holds the key to the castle and it is an in-memory structure. In case PA starts crashing say every 15 seconds, or so. So, we will not be cleaning up files that were generated in the last run as this structure is always initialized to an empty set !

Reply:
This is true. Will file this as a bug.

Reply:
So, if the reader keeps crashing every 15 seconds, there will not be much impact since the rmp waits for a I believe ~30s or longer before actually producing metricsDB files. However, I believe the issue you mention actually raises the bigger concern that metricsdb files are never cleaned up -- whether they are collected for metrics or batch metrics. As a result, if the reader crashes with the same periodicity as the metricsdb retention period (10s with just metrics processing and no batch metrics, or a peak of 60min with batch metrics) then the cluster will just accumulate a large number of metricsdb files that aren't cleaned up.

Reply:
For this initial pr, we can just cleanup any files that were found. However, we will only close ticket #415 once we've implemented initializing the batch metrics set with the available files found on startup.